hack/non-olm-install/install-gitops-operator.sh

```diff
@@ -394,6 +403,66 @@ function print_info() {
   echo "==========================================="
 }
 
+# migration from an OLM installation to a non OLM installation.
+function migrate_olm_installation() {
+  extract_custom_env_in_subscription
+  remove_subscription
+  remove_installed_csv
+  wait_for_olm_removal
+  if [ -f ${WORK_DIR}/migrate_env.sh ];then
+    echo "Sourcing env variables used for customizing subscription"
+    source ${WORK_DIR}/migrate_env.sh
+  fi
+  apply_kustomize_manifests
+  # Check pod status if it becomes ready
+  check_pod_status_ready gitops-operator-controller-manager
+}
+
+# Deletes the subscription for openshift-gitops-operator
+function remove_subscription() {
+  #Delete the gitops subscription
+  ${KUBECTL} delete subscription openshift-gitops-operator -n openshift-operators
+}
+
+# Deletes the ClusterServiceVersion Object from the system
+function remove_installed_csv() {
+  # get installedCSV from subscription status
+  installedCSV=$(${YQ} '.status.installedCSV' ${WORK_DIR}/subscription.yaml)
+  if [ "${installedCSV}" == "null" ]; then
+    echo "[INFO] No installed CSV in Subscription"
+    return
+  fi
+  ${KUBECTL} delete clusterserviceversion ${installedCSV} -n openshift-operators
+}
+
+# Waits till the OLM removal is successful.
+function wait_for_olm_removal() {
+  # Wait till the operator deployment is completely removed.
+  ${KUBECTL} wait --for=delete deploy/gitops-operator-controller-manager -n openshift-operators --timeout=60s
+}
+
+# Extract the custom configuration set in the Subscription and
+# store the env settings in a file which can be sourced when running
+# the non-OLM installation.
+function extract_custom_env_in_subscription() {
+  # Get the GitOps subscription object as yaml
+  ${KUBECTL} get subscription openshift-gitops-operator -n openshift-operators -o yaml > ${WORK_DIR}/subscription.yaml
+  # check if config.env element is present
+  element=$(${YQ} '.spec.config.env' ${WORK_DIR}/subscription.yaml)
+  if [ "${element}" == "null" ]; then
+    echo "[INFO] No custom config present in Subscription"
+    return
+  fi
+
+  # for each custom env, convert it to key=value combination.
+  while IFS=$'\t' read -r name value _; do
+    echo "Setting $name=$value"
+    echo "export $name=$value" >> ${WORK_DIR}/migrate_env.sh
+  done < <(yq e '.[] | [.name, .value] | @tsv' ${WORK_DIR}/env_overrides.yaml)
```

Comment:
Where does the `env_overrides.yaml` file get used

Reply:
This file will be used during the migration. Certain configuration settings are set in `Subscription`, like `DISABLE_DEFAULT_ARGOCD_INSTANCE` which needs to be carried over to the non OLM installation as well. The `env-overrides.yaml` file contains the env configuration set in `Subscription`